data/registry_model/registry_proxy_model.py

```diff
@@ -383,13 +383,15 @@ def _update_manifest_for_tag(
         logger.debug(f"Found upstream manifest with digest {upstream_digest}, {manifest_ref=}")
         up_to_date = manifest.digest == upstream_digest
 
+        if upstream_manifest is None:
+            upstream_manifest = self._pull_upstream_manifest(repo_ref.name, manifest_ref)
```

Comment:
moving this before the condition on line 390 means that we will pull the manifest upstream when it might not be necessary.
maybe keep this block where it was and just another check and pull the manifest if the tag is expired?
this is also why the tests are failing, `test_returns_cached_manifest_when_it_exists_upstream` ensures that we don't make an extra request upstream when one is not needed.